hermes/probe/probe.go

```diff
@@ -149,22 +140,24 @@ func (mp *Probe) runProbe(ctx context.Context, metricChan chan<- *cpmetrics.Even
 	var wg sync.WaitGroup
 	for _, t := range mp.targets {
 		wg.Add(1)
-		go func(t *Target) {
+		go func(t *target.Target) {
 			defer wg.Done()
-			var err error
-			if t.LatencyMetrics, err = metrics.NewMetrics(mp.config, t.Target); err != nil {
-				mp.logger.Errorf(err.Error())
-				return
+			if t.LatencyMetrics == nil {
+				var err error
+				if t.LatencyMetrics, err = metrics.NewMetrics(mp.config, t.Target); err != nil {
+					mp.logger.Errorf(err.Error())
+					return
+				}
 			}
+
 			probeCtx, _ := context.WithDeadline(ctx, time.Now().Add(mp.interval()))
 			start := time.Now()
```

Comment:
Can't you re-use the closure-based solution here to measure the latency?

Reply:
When I add the closure solution to metrics.go, I will refactor all of the master code to use that solution and call it from metrics.go.

Reply:
I'll add a TODO for now.